noir/noir-repo/compiler/noirc_evaluator/src/ssa/ssa_gen/program.rs

```diff
@@ -27,7 +27,29 @@ impl Ssa {
             (f.id(), f)
         });
 
-        let id_to_index = btree_map(functions.iter().enumerate(), |(i, (id, _))| (*id, i as u32));
+        let mut acir_index = 0;
+        let mut brillig_index = 0;
+        let id_to_index = btree_map(functions.iter().enumerate(), |(_, (id, func))| {
```

Comment:
This was used to get the final generated index of the ACIR entry point function. I had also wanted to avoid a shared context in ACIR gen, but now with Brillig it is unavoidable. I'm going to move to a new strategy without this map at all and just using a func id -> generated index map in the Acir gen SharedContext.

Reply:
Actually it does look easier to keep it just for ACIR functions as we do not codegen the ACIR upon making a call but separately for each SSA function. I will update with a comment.